tools/wpt/browser.py

```diff
@@ -1311,7 +1311,100 @@ def version(self, binary=None, webdriver_binary=None):
+                raise RuntimeError("Can't find a WebKitGTK MiniBrowser %s bundle for %s at %s"
+                                   % (channel, self._get_osidversion(), base_dowload_uri))
+            raise
+
+        bundle_filename = response.text.strip()
+        bundle_url = base_download_dir + bundle_filename
+
+        if dest is None:
+            dest = self._get_dest(None, channel)
+        bundle_file_path = os.path.join(dest, bundle_filename)
+
+        self.logger.info("Downloading WebKitGTK MiniBrowser bundle from %s" % bundle_url)
+        with open(bundle_file_path, "w+b") as f:
+            download_url_to_descriptor(f, bundle_url)
+
+        bundle_hash_url = bundle_url + ".sha256sum"
+        bundle_expected_hash = get(bundle_hash_url).text.strip().split(" ")[0]
+        bundle_computed_hash = sha256sum(bundle_file_path)
+
+        if bundle_expected_hash != bundle_computed_hash:
+            self.logger.error("Calculated SHA256 hash is %s but was expecting %s" % (bundle_computed_hash,bundle_expected_hash))
+            raise RuntimeError("The WebKitGTK MiniBrowser bundle at %s has incorrect SHA256 hash." % bundle_file_path)
+        return bundle_file_path
+
+    def install(self, dest=None, channel=None, prompt=True):
+        dest = self._get_dest(dest, channel)
+        bundle_path = self.download(dest, channel)
+        bundle_uncompress_directory = os.path.join(dest, "webkitgtk_minibrowser")
+
+        # Clean it from previous runs
+        if os.path.exists(bundle_uncompress_directory):
+            rmtree(bundle_uncompress_directory)
+        os.mkdir(bundle_uncompress_directory)
+
+        with open(bundle_path, "rb") as f:
+            unzip(f, bundle_uncompress_directory)
+
+        install_dep_script = os.path.join(bundle_uncompress_directory, "install-dependencies.sh")
+        if os.path.isfile(install_dep_script):
+            self.logger.info("Executing install-dependencies.sh script from bundle.")
+            install_dep_cmd = [install_dep_script]
+            if not prompt:
+                install_dep_cmd.append("--autoinstall")
+            # use subprocess.check_call() directly to display unbuffered stdout/stderr in real-time.
+            subprocess.check_call(install_dep_cmd)
+
+        minibrowser_path = os.path.join(bundle_uncompress_directory, "MiniBrowser")
+        if not os.path.isfile(minibrowser_path):
+            raise RuntimeError("Can't find a MiniBrowser binary at %s" % minibrowser_path)
+
+        os.remove(bundle_path)
+        install_ok_file = os.path.join(bundle_uncompress_directory, ".installation-ok")
+        open(install_ok_file, "w").close()  # touch
+        self.logger.info("WebKitGTK MiniBrowser bundle for channel %s installed." % channel)
+        return minibrowser_path
+
+    def _find_executable_in_channel_bundle(self, binary, venv_path=None, channel=None):
+        if venv_path:
+            venv_base_path = self._get_dest(venv_path, channel)
```

Comment:
Does this all rely on `dest` passed to `install` being the same as `venv_path`?

Reply:
Yes. It ends searching for the executable inside the folder ```_venv2/browsers/$channel/webkitgtk_minibrowser/```

Reply:
Right, it seems unfortunate if we are basically assuming that "unrelated" functions are called with the same argument. We should at the very least document that this is a requirement. Ideally we could structure things so that we don't have to rely on this assumption because it's easy to foresee it breaking if we ever allow a user-specified download location or start using e.g. a temp directory for downloads.

Reply:
Sorry, but I'm not understanding your concern here. As far as I can see the all the other browser drivers relies on this same assumption regarding the functions to install and to find the browser binaries.

For example, the function to install the firefox browser gets the main subdirectory name to install the browser binaries via the [dest parameter passed to install ](https://github.com/web-platform-tests/wpt/blob/0305367bd0926faf7a2012d4cb7f82d092d8ec2c/tools/wpt/browser.py#L215) and the function to [find the browser binary first tries to find it on that directory via the parameter venv_path](https://github.com/web-platform-tests/wpt/blob/0305367bd0926faf7a2012d4cb7f82d092d8ec2c/tools/wpt/browser.py#L263).

As far as I can see the code ensuring the parameter passed is the same is this one: [here calls install](https://github.com/web-platform-tests/wpt/blob/0305367bd0926faf7a2012d4cb7f82d092d8ec2c/tools/wpt/run.py#L168) and [here calls find_binary](https://github.com/web-platform-tests/wpt/blob/0305367bd0926faf7a2012d4cb7f82d092d8ec2c/tools/wpt/run.py#L202).

I admit this is a bit confusing and the code overall for passing the parameters to install and find_binary may benefit from a refactoring, but I think that is out of the scope of this PR.

Reply:
OK, fair enough. We should revisit this but in another PR.